shell/platform/android/io/flutter/plugin/editing/ListenableEditingState.java

```diff
@@ -179,6 +257,54 @@ public SpannableStringBuilder replace(
       Log.e(TAG, "editing state should not be changed in a listener callback");
     }
 
+    Log.e(
+        "DELTAS",
+        "replace(" + start + ", " + end + ", " + tb + ", " + tbstart + ", " + tbend + ")");
+
+    final boolean isDeletionGreaterThanOne = end - (start + tbend) > 1;
```

Comment:
If we had a TextEditingDelta class (as I mentioned above), then maybe we could encapsulate most of this logic.